### ISSUE TITLE
performance regression since 2.1.1

### ISSUE THREAD

**Issue by chafey** [team-charls/charls#103]
I just pulled the latest version of charls and notice a ~15% drop in decode performance for the WASM build vs 2.1.1 (hash 0bafe4e).  I pulled 2.2.0 and see the performance regression there as well.  Is this expected/known?  I checked the release notes and didn't see anything about this.

**Comment by vbaderks** [comment]
I was not aware that there is a performance regression on the WASM build. I don't normally profile the WASM build.
There have been changes to ensure that the fuzzer tests pass.

Any recommendation of a WASM profiler?

**Comment by vbaderks** [comment]
First checking native perf makes sense. There is no infrastructure to capture and analyze perf between releases at the moment.

The charlstest.exe tool provides an option to measure perf: -decodeperformance[:loop-count], it will look for a .jls file named decodetest.jls. Switching the charls-2-x64.dll versions should work.

The tool https://github.com/team-charls/charls-image-test will also measure decoding/encoding performance.

**Comment by vbaderks** [comment]
Initial benchmarking shows no difference between https://github.com/team-charls/charls/commit/4d1ef382beb950d3f0c2b092939a19959888fe78 and latest version for native performance:

Decoding CT1.JLS = +/- 3 ms on AMD Ryzen CPU
Decoding 8 bit gray 5412 * 7216 (big_building.jls) = +/- 510 ms on AMD Ryzen CPU

x64 release build with MSVC 19.31.30818
Note: testing done with charlstest.exe -decodeperformance:100

**Comment by vbaderks** [comment]
I compared the latest CharLS-JS (https://github.com/chafey/charls-js/commit/f5f168be90354eee0fee4d93b65906d696f7f523) with CharLS-JS + latest CharLS (cce893ef702e3ef9f85533ad9dab2980cddb67da).

When building with emscripten 3.1.1 the latest version of CharLS is even a little bit faster. For example:
CharLS-JS 2.1.1 for MG1: 221 ms
CharLS-JS 2.2.1 for MG1: 193 ms 

I saw the same type of timings for the other images and the Node benchmark tests:  equal or 2.2.1 isa little bit faster.

Note 1: With the Crome dev tools open (F12) the decoding times become much slower, sometimes even a factor of 2.
Note 2: I had to add a special `#ifdef __EMSCRIPTEN__ ` to the CharLS source code as the WebAssembly build would fail for the generic read_unaligned method.
